packages/portfolio-contract/test/rebalance.test.ts

```diff
@@ -0,0 +1,357 @@
+  {
+    srcChain: 'Avalanche',
+    destChain: 'noble',
+    variableFee: 0,
+    timeFixed: 1080,
+  },
+  { srcChain: 'Ethereum', destChain: 'noble', variableFee: 0, timeFixed: 1080 },
+  // CCTP return (Noble -> EVM)
+  { srcChain: 'noble', destChain: 'Arbitrum', variableFee: 0, timeFixed: 20 },
+  { srcChain: 'noble', destChain: 'Avalanche', variableFee: 0, timeFixed: 20 },
+  { srcChain: 'noble', destChain: 'Ethereum', variableFee: 0, timeFixed: 20 },
+  // FastUSDC unidirectional (EVM -> Noble only)
+  {
+    srcChain: 'Arbitrum',
+    destChain: 'noble',
+    variableFee: 0.0015,
+    timeFixed: 45,
+  },
+  {
+    srcChain: 'Avalanche',
+    destChain: 'noble',
+    variableFee: 0.0015,
+    timeFixed: 45,
+  },
+  {
+    srcChain: 'Ethereum',
+    destChain: 'noble',
+    variableFee: 0.0015,
+    timeFixed: 45,
+  },
+  // Added Agoric <-> Noble IBC for deposit/cash flows
+  { srcChain: 'agoric', destChain: 'noble', variableFee: 2, timeFixed: 10 },
+  { srcChain: 'noble', destChain: 'agoric', variableFee: 2, timeFixed: 10 },
+];
+
+test('solver simple 2-pool case (A -> B 30)', async t => {
+  const current = balances({ [A]: 80n, [B]: 20n });
+  const targetBps = { [A]: 5000n, [B]: 5000n };
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target: Object.fromEntries(
+      Object.entries(targetBps).map(([k, bps]) => [
+        k,
+        token((100n * bps) / 10000n),
+      ]),
+    ),
+    brand: TOK_BRAND,
+    links: LINKS,
+    mode: 'cheapest',
+  });
+
+  t.deepEqual(steps, [
+    // leaf -> hub
+    { src: A, dest: '@Arbitrum', amount: token(30n) },
+    // hub -> hub legs
+    { src: '@Arbitrum', dest: '@noble', amount: token(30n) },
+    { src: '@noble', dest: '@Avalanche', amount: token(30n) },
+    // hub -> leaf
+    { src: '@Avalanche', dest: B, amount: token(30n) },
+  ]);
+});
+
+test('solver 3-pool rounding (A -> B 33, A -> C 33)', async t => {
+  const current = balances({ [A]: 100n, [B]: 0n, [C]: 0n });
+  const targetBps = { [A]: 3400n, [B]: 3300n, [C]: 3300n };
+  const { steps } = await planRebalanceFlow({
+    assetRefs: ALL_REFS,
+    current,
+    target: Object.fromEntries(
+      Object.entries(targetBps).map(([k, bps]) => [
+        k,
+        token((100n * bps) / 10000n),
+      ]),
+    ),
+    brand: TOK_BRAND,
+    links: LINKS, // fixed: was a string literal
+    mode: 'cheapest',
+  });
+
+  const amt66 = token(66n);
+  const amt33 = token(33n);
+  t.deepEqual(steps, [
+    // leaf -> hub (aggregated outflow from A)
+    { src: A, dest: '@Arbitrum', amount: amt66 },
+    // hub -> hub aggregated then split
+    { src: '@Arbitrum', dest: '@noble', amount: amt66 },
+    { src: '@noble', dest: '@Avalanche', amount: amt33 },
+    { src: '@noble', dest: '@Ethereum', amount: amt33 },
+    // hub -> leaf
+    { src: '@Avalanche', dest: B, amount: amt33 },
+    { src: '@Ethereum', dest: C, amount: amt33 },
+  ]);
```

Comment:
I think steps like this would make it harder for the UI to show to the users because there's no information about the transfers

Reply:
These are the steps that the current contract code expects.